src/crypto/index.js

```diff
@@ -1052,79 +1063,117 @@ Crypto.prototype._onRoomKeyRequestEvent = function(event) {
  *
  * @private
  */
-Crypto.prototype._processReceivedRoomKeyRequests = function() {
-    const requests = this._receivedRoomKeyRequests;
-    this._receivedRoomKeyRequests = [];
-    for (const req of requests) {
-        const userId = req.userId;
-        const deviceId = req.deviceId;
+Crypto.prototype._processReceivedRoomKeyRequests = async function() {
+    if (this._processingRoomKeyRequests) {
+        // we're still processing last time's requests; keep queuing new ones
+        // up for now.
+        return;
```

Comment:
https://github.com/matrix-org/matrix-js-sdk/blob/89e649f51f7f0e2668cab0e89e9ac9814f890196/src/crypto/index.js#L1053

This whole function is about processing that queue, so I don't think it merits making this comment more explicit.